Test/VimCoreTest/InterpreterTest.cs

```diff
@@ -1470,6 +1470,23 @@ public void RHSCanBeStringNotEqualExpression()
                 AssertValue("x", 1);
             }
 
+            [WpfFact]
+            public void LHSCanBeEnvironmentVariable()
+            {
+                Create("");
+                var variable = "magic";
+                var value = "xyzzy";
+                try
+                {
+                    ParseAndRun($"let ${variable} = '{value}'");
+                    Assert.Equal(value, System.Environment.GetEnvironmentVariable(variable));
+                }
+                finally
+                {
+                    Environment.SetEnvironmentVariable(variable, null);

```

Comment:
Really minor nit: As unlikely as it is that someone would define `%magic%` i'm sure there is someone who does it. Consider saving and restoring the original value here vs. nulling it out.

Reply:
Yes, I was thinking about this. Not corrupting the process environment is probably good etiquette, if not strictly necessary.

The only possible negative consequence to manipulating the environment is interfering with another test, and the only way to prevent that is to ensure that tests that manipulate the environment like this are run sequentially rather than in parallel. I believe **all** our tests run sequentially now so even restoring it is unnecessary. It would be better if we could annotate the test as non-parallelizable, but I don't know if that formality is necessary at the moment.

I'll restore the former value and make a comment about running tests in parallel.